File: Level 2/기능개발/solution.js
function solution(progresses, speeds) {
  const answer = [];

  while (progresses.length !== 0) {
    const progressedProgresses = progresses.map(
      (progress, index) => progress + speeds[index]
    );
    let completedFeature = 0;

    while (progressedProgresses[0] >= 100) {
      progressedProgresses.shift();
      speeds.shift();
      completedFeature++;
    }

    progresses = progressedProgresses;

    if (completedFeature > 0) {
      answer.push(completedFeature);
    }
  }

  return answer;
}

File: Level 2/기능개발/solution.py
import math


def solution(progresses, speeds):
    answer = []

    count = 0
    max_day = None

    for progress, speed in zip(progresses, speeds):
        day_left = math.ceil((100 - progress) / speed)

        if max_day is None:
            max_day = day_left

        if day_left <= max_day:
            count += 1
        else:
            answer.append(count)
            count = 1
            max_day = day_left

    answer.append(count)

    return answer
